src/katsdpmodels/primary_beam.py

```diff
@@ -234,7 +323,437 @@ def sample_grid(self, l: ArrayLike, m: ArrayLike, frequency: u.Quantity,   # noq
         Refer to :meth:`sample` for further details of the parameters and
```

Comment:
The method abstract suggests the grid must be regular, but the extended abstract contradicts this.

Speaking of irregular grids, might we not run into trouble with, for instance, a Fourier representation of the PB?

Reply:
I've changed it from "regular grid" to "grid aligned to the axes".

> Speaking of irregular grids, might we not run into trouble with, for instance, a Fourier representation of the PB?

The Fourier relationship benefits even from an irregularly-spaced but axis-aligned grid, because the Fourier transform is a separable filter. The *fast* Fourier transform would benefit from a regularly-spaced grid, but only for the right step sizes, and the primary use case for this function is in imaging where other considerations (resolution of the synthesized beam) determine the pixel sizes.

One could nevertheless consider a `sample_regular_grid`  convenience function whose default implementation is simply to call `sample_grid`. I think that should be a new PR though, since it's changing the API rather than implementing it.